sources/packages/backend/apps/api/src/services/student-file/student-file.service.ts

```diff
@@ -72,6 +73,10 @@ export class StudentFileService extends RecordDataModelService<StudentFile> {
     newFile.student = { id: studentId } as Student;
     newFile.creator = { id: auditUserId } as User;
     newFile.virusScanStatus = VirusScanStatus.InProgress;
+    // Generate the file hash.
+    newFile.fileHash = createHash("sha256")
+      .update(createFile.fileContent)
+      .digest("hex");
```

Comment:
Would it be better if we consider the file name also when creating the SIN hash?

Reply:
The unique file name is exactly what we need to prevent to cause any change in the dash.
The idea here is to hash the file content only to allow the file content only verification.

Reply:
I got the intention now which follows the AC. Thanks for the sync up.